### PR TITLE
Feature/case worker details

### PR DESCRIPTION
**Pull request by udaydenduluri33**
https://tools.hmcts.net/jira/browse/EUI-2998
https://tools.hmcts.net/jira/browse/EUI-3023
https://tools.hmcts.net/jira/browse/EUI-3014

Handling File upload Unhappy path and partial success scenarios.

Does this PR introduce a breaking change? (check one with "x")

[ ] Yes
[ x ] No

### CODE REVIEW

File: api/caseWorkerDetailsRouter/index.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import * as express from 'express'
+import * as multer from 'multer'
+import { getErrorResponse, getPartialSuccess, getSuccess } from './mock-error-scenarios'
+
+const upload = multer({ dest: 'uploads/' })
+
+async function caseWorkerDetailsRoute(req: express.Request, res: express.Response) {
+    const mockErrorRes = getErrorResponse(req, res)
```
Comment (paul-graham) [inline api/caseWorkerDetailsRouter/index.ts:8]: Indent of 4 spaces throughout this file. Should be 2.

File: api/caseWorkerDetailsRouter/index.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    
+    if(mockErrorRes) {
+        res.status(mockErrorRes.status).json(mockErrorRes.json)
+        return
+    }
+
+    const partialSuccessRes = getPartialSuccess(req, res)
+    if(partialSuccessRes) {
+        res.send(partialSuccessRes)
+        return
```
Comment (paul-graham) [inline api/caseWorkerDetailsRouter/index.ts:18]: This is redundant. Nothing happens afterwards to warrant a return at this point.

File: api/caseWorkerDetailsRouter/mock-error-scenarios.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
+
```
Comment (paul-graham) [inline api/caseWorkerDetailsRouter/mock-error-scenarios.ts:1]: Seeing as this is a mock, it should be called `mock-error-scenarios.spec.ts` or else it'll be included in code coverage. Or is this a temporary thing?

File: src/case-worker-ref-data/components/case-worker-details/case-worker-details.component.html [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+                    <div class="govuk-form-group" [ngClass]="{ 'govuk-form-group--error': errorDesc }" id="uploadarea">
+                        <label class="govuk-label" for="file-upload-1"><strong>
+                            Upload file</strong>
+                        </label>
+
+                        
+                            <span id="" *ngIf="errorDesc" class="govuk-error-message">
+                              <span class="govuk-visually-hidden">Error:</span> {{errorDesc}}
+                            </span>
+                    <input class="govuk-file-upload" #Excel id="file-upload-1" name="file-upload-1" type="file" accept=" .xls, .xlsx">
```
Comment (paul-graham) [inline src/case-worker-ref-data/components/case-worker-details/case-worker-details.component.html:38]: Indenting is off on this line. The whole file should be 2 space indenting, but this one is further back than it should be.

File: src/case-worker-ref-data/components/upload-info-details/upload-info-details.component.html [hunk 5]
```diff
@@ -0,0 +1,26 @@
+<div class="hmcts-width-container">
+    <main class="govuk-main-wrapper " id="main-content" role="main">
+        <div class="govuk-grid-row">
+            <div class="govuk-grid-column-two-thirds">
+                <div class="govuk-panel govuk-panel--confirmation">
+                    <h1 class="govuk-panel__title">
+                        Upload successful
+                    </h1>
+                    <div class="govuk-panel__body">
+                        <br><strong></strong>
```
Comment (paul-graham) [inline src/case-worker-ref-data/components/upload-info-details/upload-info-details.component.html:10]: What is the `<strong></strong>` for?

File: src/case-worker-ref-data/components/upload-info-details/upload-info-details.component.html [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+                </div>
+                <p class="govuk-body"></p>
+                <p class="govuk-body"> {{uploadResponse.message_details}}<br>
+                </p>
+                <h2 class="govuk-heading-m">What happens next</h2>
+                <p class="govuk-body">
+                    You’ll now be able to assign tasks to your colleagues <a href="https://manage-case.platform.hmcts.net/" target="_">using the Manage cases application</a>.
+                </p>
+                <p class="govuk-body">
+                <a href="#" class="govuk-link"></a>
```
Comment (paul-graham) [inline src/case-worker-ref-data/components/upload-info-details/upload-info-details.component.html:21]: Needs an extra indent.

File: src/case-worker-ref-data/components/upload-info-partial-success/upload-info-partial-success.html [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+                <h1 class="govuk-heading-l">Some of the records weren’t updated </h1>
+                <p class="govuk-body">
+                    {{partialErrorResp.message_details}}
+                </p>
+                <h3 class="govuk-heading-m">What went wrong </h3>
+                <p class="govuk-body">The file uploaded with the following errors:</p>
+                <table class="govuk-table">
+                    <thead class="govuk-table__head">
+                        <tr class="govuk-table__row">
+                        <th scope="col" class="govuk-table__header">Row</th>
```
Comment (paul-graham) [inline src/case-worker-ref-data/components/upload-info-partial-success/upload-info-partial-success.html:14]: These need indenting.

File: src/case-worker-ref-data/utils/case-worker-utils.ts [hunk 8]
```diff
@@ -0,0 +1,60 @@
+import { NavigationExtras } from '@angular/router';
```
Comment (paul-graham) [inline src/case-worker-ref-data/utils/case-worker-utils.ts:1]: `case-worker` should be `caseworker` as it's a single unhyphenated word.

**Review by paul-graham — CHANGES_REQUESTED** [review]

File: api/caseWorkerDetailsRouter/mock-error-scenarios.ts (hunk 3 above)
Comment (connorpgpmcelroy) [reply api/caseWorkerDetailsRouter/mock-error-scenarios.ts:1]: As discussed, keeping as is due to usage within code and temporary lifespan

File: src/caseworker-ref-data/components/caseworker-details/caseworker-details.component.ts [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+
+@Component({
+  selector: 'app-prd-caseworker-details',
+  templateUrl: './caseworker-details.component.html',
+})
+export class CaseWorkerDetailsComponent {
+
+  public errorDesc: string;
+
+  public constructor(private readonly caseWorkerRefDataService: CaseWorkerRefDataService,
```
Comment (paul-graham) [inline src/caseworker-ref-data/components/caseworker-details/caseworker-details.component.ts:15]: This looks wrong. That first parameter should also be on a new line. I think lint might complain about this.

**Review by paul-graham — CHANGES_REQUESTED** [review]

**Review by paul-graham — APPROVED** [review]

**Review by connorpgpmcelroy — APPROVED** [review]

**Review by Mo-Lala84 — APPROVED** [review]
lgtm approved.

**Pull request merged**